crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -0,0 +1,1094 @@
+                )
+                .unwrap()
+                .a;
+            let mut fidelity_value = match available_2q_props.get(name) {
+                Some(&(_, error)) => 1.0 - error.unwrap_or(0.0),
+                None => 1.0,
+            };
+            if let Some(approx_degree) = approximation_degree {
+                fidelity_value *= approx_degree;
+            }
+            let mut embodiment =
+                xx_embodiments.get_item(op.clone().into_py(py).getattr(py, "base_class")?)?; //XXEmbodiments[v.base_class];
+
+            if embodiment.getattr("parameters")?.len()? == 1 {
+                embodiment = embodiment.call_method1("assign_parameters", (vec![strength],))?;
+            }
+            // basis equivalent to CX are well optimized so use for the pi/2 angle if available
+            if relative_eq!(strength, PI2) && supercontrolled_basis.contains_key(name) {
+                pi2_basis = Some(op.operation.name());
+            }
+            Ok((strength, fidelity_value, embodiment))
+        },
+    );
+
+    let basis_2q_fidelity_dict = PyDict::new_bound(py);
+    let embodiments_dict = PyDict::new_bound(py);
+    let mut not_empty = false;
+    for (strength, fidelity, embodiment) in xx_decomposer_args.flatten() {
+        not_empty = true;
+        basis_2q_fidelity_dict.set_item(strength, fidelity)?;
+        embodiments_dict.set_item(strength, embodiment.into_py(py))?;
+    }
+
+    // Iterate over 2q fidelities ans select decomposers
+    if not_empty {
+        let xx_decomposer: Bound<'_, PyAny> =
+            PyModule::import_bound(py, "qiskit.synthesis.two_qubit.xx_decompose")?
+                .getattr("XXDecomposer")?;
+
+        for basis_1q in &available_1q_basis {
+            let pi2_decomposer = if let Some(pi_2_basis) = pi2_basis {
+                if pi_2_basis == "cx" && *basis_1q == "ZSX" {
+                    let fidelity = match approximation_degree {
+                        Some(approx_degree) => approx_degree,
+                        None => match &target["cx"][Some(&qubits)] {
+                            Some(props) => 1.0 - props.error.unwrap_or(0.0),
+                            None => 1.0,
+                        },
+                    };
+                    Some(TwoQubitBasisDecomposer::new_inner(
+                        pi_2_basis.to_string(),
+                        StandardGate::CXGate.matrix(&[]).unwrap().view(),
+                        fidelity,
+                        basis_1q,
+                        Some(true),
+                    )?)
+                } else {
+                    None
+                }
+            } else {
+                None
+            };
+
+            let decomposer = xx_decomposer.call1((
+                &basis_2q_fidelity_dict,
+                PyString::new_bound(py, basis_1q),
+                &embodiments_dict,
+                pi2_decomposer,
+            ))?;
+            let decomposer_gate = decomposer.getattr("gate")?;
```

Comment:
To save building a python string `gate` on each call to this you can do:
```suggestion
            let decomposer_gate = decomposer.getattr(intern!(py, "gate"))?;
```